src/dependency-manager/src/manager.ts

```diff
@@ -127,13 +126,19 @@ export default abstract class DependencyManager {
     }
 
     // Load component dependencies
-    if (load_dependencies) {
-      for (const [dep_key, dep_value] of Object.entries(component.getDependencies())) {
-        const dep_name = dep_value.includes(':') ? `${dep_key}:latest` : `${dep_key}:${dep_value}`;
-        const dep_extends = dep_value.includes(':') ? dep_value : `${dep_key}:${dep_value}`;
-        const dep_component = ComponentConfigBuilder.buildFromJSON({ extends: dep_extends, name: dep_name });
-        await this.loadComponent(graph, dep_component, component_map);
+    for (const [dep_key, dep_value] of Object.entries(component.getDependencies())) {
+      const dep_name = dep_value.includes(':') ? `${dep_key}:latest` : `${dep_key}:${dep_value}`;
+      const dep_extends = dep_value.includes(':') ? dep_value : `${dep_key}:${dep_value}`;
+      const dep_component = ComponentConfigBuilder.buildFromJSON({ extends: dep_extends, name: dep_name });
+
+      if (component_map[dep_component.getRef()]) {
+        const first_side_dependency = ComponentVersionSlugUtils.parse(component.getRef()).namespaced_component_name in component_map[dep_component.getRef()].getDependencies();
+        const second_side_dependency = ComponentVersionSlugUtils.parse(dep_component.getRef()).namespaced_component_name in component.getDependencies();
```

Comment:
Pretty sure `second_side_dependency` is redundant since you're looping over the component dependencies.